### PR TITLE
changed to xenial

### PR DESCRIPTION
**Pull request by JonathanPortillo**

### CODE REVIEW

**Comment by clonm** [comment]
Based on more testing with docker-blog, it looks like this doesn't completely work yet. 
* I'm not sure why it ever worked without a password, but at least with this version, it shuts down if we don't set one manually. To fix this, I tried adding a password in [wp-config.php](https://github.com/cloyne/docker-blog/blob/master/development/etc/wordpress/wp-config.php) by changing `define('DB_PASSWORD', '');` to `define('DB_PASSWORD', 'dev-only');`,
and in [docker-mysql-dev](https://github.com/cloyne/docker-mysql-dev) by adding `sed -i 's/password.*=.*/password = dev-only/' /etc/mysql/debian.cnf`. I haven't committed these yet because in the mysql-dev one, that's an autogenerated file that's not supposed to be edited manually, so there is probably a better way.
* With a default password configured, it keeps crashing and restarting. Haven't figured out why yet but probably has something to do with runit on Xenial.
```
190729 07:00:56 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
190729  7:00:56 [Note] /usr/sbin/mysqld (mysqld 10.0.38-MariaDB-0ubuntu0.16.04.1) starting as process 209 ...
190729  7:00:56 [Note] InnoDB: innodb_empty_free_list_algorithm has been changed to legacy because of small buffer pool size. In order to use backoff, increase buffer pool at least up to 20MB.

190729  7:00:56 [Note] InnoDB: Using mutexes to ref count buffer pool pages
190729  7:00:56 [Note] InnoDB: The InnoDB memory heap is disabled
190729  7:00:56 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
190729  7:00:56 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
190729  7:00:56 [Note] InnoDB: Compressed tables use zlib 1.2.8
190729  7:00:56 [Note] InnoDB: Using Linux native AIO
190729  7:00:56 [Note] InnoDB: Using CPU crc32 instructions
190729  7:00:56 [Note] InnoDB: Initializing buffer pool, size = 128.0M
190729  7:00:56 [Note] InnoDB: Completed initialization of buffer pool
190729  7:00:56 [Note] InnoDB: Highest supported file format is Barracuda.
190729  7:00:56 [Note] InnoDB: 128 rollback segment(s) are active.
190729  7:00:56 [Note] InnoDB: Waiting for purge to start
190729  7:00:56 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.42-84.2 started; log sequence number 1616717
190729  7:00:56 [Note] Plugin 'FEEDBACK' is disabled.
190729  7:00:56 [Note] Server socket created on IP: '127.0.0.1'.
190729  7:00:56 [Note] /usr/sbin/mysqld: ready for connections.
Version: '10.0.38-MariaDB-0ubuntu0.16.04.1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  Ubuntu 16.04
190729  7:00:57 [Note] /usr/sbin/mysqld: Normal shutdown
190729  7:00:57 [Note] InnoDB: FTS optimize thread exiting.
190729  7:00:57 [Note] InnoDB: Starting shutdown...
190729  7:00:57 [Note] InnoDB: Waiting for page_cleaner to finish flushing of buffer pool
190729  7:00:59 [Note] InnoDB: Shutdown completed; log sequence number 1616727
190729  7:00:59 [Note] /usr/sbin/mysqld: Shutdown complete

190729 07:00:59 mysqld_safe mysqld from pid file /var/run/mysqld/mysqld.pid ended
190729 07:00:59 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
190729  7:00:59 [Note] /usr/sbin/mysqld (mysqld 10.0.38-MariaDB-0ubuntu0.16.04.1) starting as process 486 ...

```
etc. until stopped.

**Comment by mitar** [comment]
Have you checked if that patch cleanly applies?

File: mysql-5.5.dockerfile [hunk 1]
```diff
@@ -8,18 +7,18 @@ VOLUME /var/log/mysql
 VOLUME /var/lib/mysql
 
 RUN apt-get update -q -q && \
- apt-get install --yes --force-yes mariadb-server && \
```
Comment (mitar) [inline mysql-5.5.dockerfile:11]: Let's not remove this for now.

**Comment by mitar** [comment]
Try out the latest version.

**Comment by clonm** [comment]
The latest version worked, mostly.

To upgrade `mysql`, I stopped the mysql service with `sv stop mysql` inside the container, then (from outside the container) backed up the entire folder /srv/mysql to /srv/mysql.bak. I then then stopped the container and renamed it, then pulled the latest `tozd/mysql` and ran it with ` docker run --volume /srv/mysql:/var/lib/mysql --volume /srv/var/log/mysql:var/log/mysal --network=server2.cloyne.org --name "mysql" --hostname "mysql" --detach true clonm/mysql:10.1`. 

Then, inside the container, ran `mysql_upgrade -p` and input the database password at the prompt. I had to fix the permissions by running `chown mysql:mysql -R /var/lib/mysql`. Restarting with `sv restart mysql` timed out, so I restarted the container instead with `docker restart mysql` outside the container.

After these fixes, the server ran successfully but nothing could connect to it. This is because the new default `bind_address` is `127.0.0.1`, and this is set in a later config file than where we set it to `0.0.0.0` in `local.cnf`. I hotfixed this by swapping the last two lines in `/etc/mysql/mariadb.cnf`, so now it ends with
```
!includedir /etc/mysql/mariadb.conf.d/
!includedir /etc/mysql/conf.d/
```
But this is not currently reflected in the image. It would be preferable to just copy the `local.cnf` file to `/etc/mysql/mariadb.conf.d/60-local.cnf`, but this seems to be overwritten on startup. I will make a patch once I have a more stable internet connection to test docker builds.

**Comment by mitar** [comment]
The image has been updated now to a range of (working) versions of MySQL/MairaDB. It also includes CI tests to make sure things work.

**Pull request closed without merging**